cmd/power/src/lib.rs

```diff
@@ -28,15 +35,84 @@ struct PowerArgs {
         parse(try_from_str = parse_int::parse)
     )]
     timeout: u32,
+
+    /// get phase current where available
+    #[clap(long)]
+    phase_current: bool,
 }
 
 struct Device<'a> {
     name: &'a str,
+    rail: Option<usize>,
     voltage: Option<usize>,
     current: Option<usize>,
     input_voltage: Option<usize>,
     input_current: Option<usize>,
     temperature: Option<usize>,
+    phases: Option<&'a Vec<u8>>,
+    phase_currents: Option<Vec<Option<f32>>>,
+}
+
+fn phase_currents(
+    hubris: &HubrisArchive,
+    core: &mut dyn Core,
+    context: &mut HiffyContext,
+    devices: &mut BTreeMap<(&String, HubrisSensorDevice), Device>,
+) -> Result<()> {
+    let op = hubris.get_idol_command("Power.phase_current")?;
+    let ok = hubris.lookup_basetype(op.ok)?;
+    let mut ops = vec![];
+
+    if ok.encoding != HubrisEncoding::Float {
+        bail!("expected return value of phase_current() to be a float");
+    }
+
+    if ok.size != 4 {
+        bail!("expected return value of phase_current() to be an f32");
+    }
+
+    for device in devices.values() {
+        if let Some(phases) = device.phases {
+            let id = match device.rail {
+                Some(id) => id,
+                None => {
+                    bail!("{} does not have a voltage sensor?", device.name);
+                }
+            };
+
+            for (phase, _) in phases.iter().enumerate() {
+                let payload = op.payload(&[
+                    ("rail", idol::IdolArgument::Scalar(id as u64)),
+                    ("phase", idol::IdolArgument::Scalar(phase as u64)),
+                ])?;
+                context.idol_call_ops(&op, &payload, &mut ops)?;
+            }
+        }
+    }
+
+    ops.push(Op::Done);
+    let results = context.run(core, ops.as_slice(), None)?;
+
+    let mut ndx = 0;
+
+    for device in devices.values_mut() {
+        if let Some(phases) = device.phases {
+            let mut phase_currents = vec![];
+
+            for _ in phases.iter() {
+                phase_currents.push(match &results[ndx] {
+                    Ok(val) => Some(f32::from_le_bytes(val[0..4].try_into()?)),
+                    _ => None,
```

Comment:
Should we be logging something about the error here?

Reply:
Fixed by using the error character ("x") rather than the absence character ("-") when printing the phase.